### PR TITLE
add and remove questions from questionpool

### PR DESCRIPTION
**Pull request by FelixDittrich**

### CODE REVIEW

File: Frontend/SPFrontend/src/app/app-routing.module.ts [hunk 1]
```diff
@@ -12,7 +12,7 @@ const routes: Routes = [
   {path: 'home', component: HomePageComponent, canActivate: [AuthGuard]},
   {path: 'login', component: LoginComponent},
   {path:'questions-overview', component: QuestionsOverviewComponent, canActivate: [AuthGuard]},
-  {path: 'questions-collection', component: QuestionsCollectionComponent,canActivate: [AuthGuard]},
+  {path: 'questions-collection', component: QuestionsCollectionComponent},
```
Comment (MediocreMarcel) [inline Frontend/SPFrontend/src/app/app-routing.module.ts]: Do not remove authguard

File: Frontend/SPFrontend/src/app/components/models/QuestionDto.ts [hunk 2]
```diff
@@ -1,21 +1,20 @@
 export class QuestionDto {
 
-  questionId: number;
+  //questionId: number;
```
Comment (MediocreMarcel) [inline Frontend/SPFrontend/src/app/components/models/QuestionDto.ts]: do not remove

File: Frontend/SPFrontend/src/app/components/models/QuestionDto.ts [hunk 3]
```diff
@@ -1,21 +1,20 @@
...
   questionName: string;
   questionText: string;
   questionPoints: number;
   shortName: string;
   category: string;
   module_ID: number;
 
-  constructor(questionId:number, questionName:string, questionText:string,questionPoints:number,shortName:string, category:string, module_ID:number) {
-    this.questionId = questionId;
+  constructor(questionName: string, questionText: string, questionPoints: number, shortName: string, category: string, module_ID: number) {
```
Comment (MediocreMarcel) [inline Frontend/SPFrontend/src/app/components/models/QuestionDto.ts]: do not remove

File: Frontend/SPFrontend/src/app/components/models/QuestionDto.ts (hunk 2 above)
Comment (ghost) [inline Frontend/SPFrontend/src/app/components/models/QuestionDto.ts]: do not remove id

File: Frontend/SPFrontend/src/app/services/question/create-question.service.ts [hunk 4]
```diff
@@ -22,7 +22,7 @@ export class CreateQuestionService {
   writeQuestionToDb(postData: QuestionDto){
     const headers = new HttpHeaders().set('Content-Type', 'application/json; charset=utf-8');
     console.log(postData);
-    return this.http.post<QuestionDto>(this.url + 'questions/newQuestion', JSON.stringify(postData), {headers: headers});
+    return this.http.post<QuestionDto>(this.url + 'questions/newQuestion', JSON.stringify(postData), {headers});
```
Comment (MediocreMarcel) [inline Frontend/SPFrontend/src/app/services/question/create-question.service.ts]: undo change

File: Frontend/SPFrontend/src/app/services/question/create-question.service.ts [hunk 5]
```diff
@@ -22,7 +22,7 @@ export class CreateQuestionService {
   writeQuestionToDb(postData: QuestionDto){
     const headers = new HttpHeaders().set('Content-Type', 'application/json; charset=utf-8');
     console.log(postData);
```
Comment (MediocreMarcel) [inline Frontend/SPFrontend/src/app/services/question/create-question.service.ts]: remove log

File: Frontend/SPFrontend/src/app/components/questions-collection/questions-collection.component.ts [hunk 6]
```diff
@@ -73,3 +82,27 @@ export class QuestionsCollectionComponent implements OnInit {
...
+  shortName: string;
+  questionText: string;
+  questionPoints: number;
+  category: string;
+  module: ModuleDTO;
+
+  constructor(private service: CreateQuestionService, private dialogRef: MatDialogRef<CreateQuestionDialog>) {
+  }
+
+  writeQuestion() {
```
Comment (MediocreMarcel) [inline Frontend/SPFrontend/src/app/components/questions-collection/questions-collection.component.ts:115]: add /** comment

File: Frontend/SPFrontend/src/app/components/questions-collection/questions-collection.component.ts [hunk 7]
```diff
@@ -64,7 +65,15 @@ export class QuestionsCollectionComponent implements OnInit {
   }
 
   createQuestion() {
```
Comment (MediocreMarcel) [inline Frontend/SPFrontend/src/app/components/questions-collection/questions-collection.component.ts:73]: add /**

File: Frontend/SPFrontend/src/app/components/questions-collection/questions-collection.component.ts [hunk 8]
```diff
@@ -64,7 +65,15 @@ export class QuestionsCollectionComponent implements OnInit {
...
+      height: '80%'
+    });
+
+    dialogRef.afterClosed().subscribe(create => {
+      this.loadQuestions();
+      console.log(this.questions);
+    });
   }
 
   loadQuestions() {
```
Comment (MediocreMarcel) [inline Frontend/SPFrontend/src/app/components/questions-collection/questions-collection.component.ts:88]: add /** comment

File: Frontend/SPFrontend/src/app/components/dto/questionDto.ts [hunk 9]
```diff
@@ -0,0 +1,18 @@
+export class QuestionDto {
```
Comment (MediocreMarcel) [inline Frontend/SPFrontend/src/app/components/dto/questionDto.ts]: delete file

**Review by MediocreMarcel — CHANGES_REQUESTED** [review]
fix changes

File: Frontend/SPFrontend/src/app/components/questions-collection/questions-collection.component.html [hunk 10]
```diff
@@ -70,7 +70,7 @@
     </mat-card>
 
   </mat-card-content>
-  <mat-card-footer> copyright
+  <mat-card-footer>
```
Comment (ghost) [inline Frontend/SPFrontend/src/app/components/questions-collection/questions-collection.component.html]: footer komplett entfernen

**Pull request merged**